client.go

```diff
@@ -217,11 +217,9 @@ func (cs *clientStream) RecvMsg(m interface{}) error {
 	select {
 	case <-cs.ctx.Done():
 		return cs.ctx.Err()
-	case msg, ok := <-cs.s.recv:
-		if !ok {
-			return cs.s.recvErr
-		}
-
+	case <-cs.s.recvClose:
+		return cs.s.recvErr
+	case msg := <-cs.s.recv:
```

Comment:
Is it necessary to ensure that messages that have been sent to recv are processed?

```go
var msg *streamMessage

select {
...
case <- cs.s.recvClose:
    select {
        case msg <- cs.s.recv:
        default:
            return cs.s.recvErr
    }
case msg <- cs.s.recv:
}
if msg.header.Type == messageTypeResponse {
...
```

Because there is a check before receiving data, s.recv will not always have data

Reply:
Updated, that is a good change. Wish select could have ordering which would make code much cleaner